modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -325,4 +303,89 @@ public function get_user_id() {
 		return 'null';
 	}
 
+	/**
+		 * @param $cart_item_key
+		 * @param $product_id
+		 * @param $quantity
+		 * @param $variation_id
+		 * @param $variation
+		 * @param $cart_item_data
+		 */
+		public function capture_add_to_cart( $cart_item_key, $product_id, $quantity, $variation_id, $variation, $cart_item_data ) {
+			$referer_postid = isset( $_SERVER['HTTP_REFERER'] ) ? url_to_postid( $_SERVER['HTTP_REFERER'] ) : 0;
+			// if the referring post is not a product OR the product being added is not the same as post
+			// (eg. related product list on single product page) then include a product view event
```

Comment:
Also noting from our convo today - how will this bit of logic respond to a quick add event from a product listing page?

Reply:
Good point. It captures the `add_to_cart` action properly -- but it does also fire off a view which might not be accurate in this case.

Reply:
yes, adding a product from the product page results in 2 `product_view`s and one `add-to-cart`

>if the referring post is not a product OR the product being added is not the same as post
(eg. related product list on single product page) then include a product view event

This is correct, and when I step through the code, this portion behaves as expected. Digging further, the extra page view is from the page refresh, and you can see that working in the master branch.

@greenafrican Since we are utilizing `session`, in `capture_product_view` we can check to see if a `product_view` is about to be dropped. This may depend on timing. Is it worth exploring?

Reply:
> Digging further, the extra page view is from the page refresh

Yes, I had an extra check to prevent these but removed it yesterday as it can cause other issues down the line. For example, if the user continues to `add_to_cart` we log all the additional `add_to_cart` events but no further `product_views`. It is not ideal/perfect as is but I'd rather there were a few extra `product_views` than have a situation where there can be more `add_to_carts` than `product_views`. The more I think about this the more I wonder if we don't need to move to reporting on uniques rather than simple counts.

Reply:
> how will this bit of logic respond to a quick add event from a product listing page?

@timmyc It will add both a `product_view` and an `add_to_cart` event, which is what we want. 

> It captures the add_to_cart action properly -- but it does also fire off a view which might not be accurate in this case.

@justinshreve My thoughts are that we do want the `product_view` here for 2 reasons: 1) we don't want a situation where we have more `add_to_cart` events than `product_views` in a funnel, 2) I feel as though, by clicking `add_to_cart`, the user has 'looked' at the product.